src/org/sosy_lab/java_smt/solvers/opensmt/OpenSmtFormulaCreator.java

```diff
@@ -0,0 +1,462 @@
+import org.sosy_lab.java_smt.basicimpl.FormulaCreator;
+import org.sosy_lab.java_smt.basicimpl.FunctionDeclarationImpl;
+import org.sosy_lab.java_smt.solvers.opensmt.OpenSmtFormula.OpenSmtArrayFormula;
+import org.sosy_lab.java_smt.solvers.opensmt.OpenSmtFormula.OpenSmtBooleanFormula;
+import org.sosy_lab.java_smt.solvers.opensmt.OpenSmtFormula.OpenSmtIntegerFormula;
+import org.sosy_lab.java_smt.solvers.opensmt.OpenSmtFormula.OpenSmtRationalFormula;
+
+public class OpenSmtFormulaCreator extends FormulaCreator<PTRef, SRef, Logic, SymRef> {
+
+  private Logics logicToUse;
+
+  private OpenSmtFormulaCreator(Logics logicType, Logic logic) {
+    super(
+        logic,
+        logic.getSort_bool(),
+        (logic instanceof ArithLogic) ? ((ArithLogic) logic).getSort_int() : null,
+        (logic instanceof ArithLogic) ? ((ArithLogic) logic).getSort_real() : null,
+        null,
+        null);
+
+    logicToUse = logicType;
+  }
+
+  public static OpenSmtFormulaCreator newCreator(Logics logicType) {
+    switch (logicType) {
+      case CORE:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getInstance(Logic_t.QF_BOOL));
+      case QF_AX:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getInstance(Logic_t.QF_AX));
+      case QF_UF:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getInstance(Logic_t.QF_UF));
+      case QF_IDL:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_IDL));
+      case QF_RDL:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_RDL));
+      case QF_LIA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_LIA));
+      case QF_LRA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_LRA));
+      case QF_ALIA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_ALIA));
+      case QF_ALRA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_ALRA));
+      case QF_UFLIA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_UFLIA));
+      case QF_UFLRA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_UFLRA));
+      case QF_AUFLIA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_AUFLIA));
+      case QF_AUFLRA:
+        return new OpenSmtFormulaCreator(logicType, LogicFactory.getLAInstance(Logic_t.QF_AUFLRA));
+      default:
+        return new OpenSmtFormulaCreator(Logics.ALL, LogicFactory.getLogicAll());
+    }
+  }
+
+  boolean hasArrays() {
+    List<Logics> supported =
+        Arrays.asList(
+            Logics.QF_AX,
+            Logics.QF_ALIA,
+            Logics.QF_ALRA,
+            Logics.QF_AUFLIA,
+            Logics.QF_AUFLRA,
+            Logics.ALL);
+
+    return supported.contains(logicToUse);
```

Comment:
Linear runtime from List. Set would be constant runtime.
Here ,this is negligible due to size, but keep in mind for other situations.